common/src/main/scala/net/psforever/objects/zones/ZonePopulationActor.scala

```diff
@@ -63,13 +82,28 @@ class ZonePopulationActor(zone : Zone, playerMap : TrieMap[Avatar, Option[Player
       }
 
     case Zone.Corpse.Add(player) =>
-      CorpseAdd(player, corpseList)
+      val (playerIsCorpse, playerInZone) = CorpseAdd(player, playerMap, corpseList)
+      if(playerIsCorpse) {
+        if(!playerInZone) {
```

Comment:
It's a really minor gripe, but you could combine the two if statements to reduce nesting